From 7f409e401bd263ab2abd9b1edc0c363295c74313 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:53 -0400
Subject: [usb] uas: Add some data in/out ready iu sanity checks

Message-id: <1410887463-20674-342-git-send-email-dzickus@redhat.com>
Patchwork-id: 95294
O-Subject: [RHEL7 PATCH 341/591] uas: Add some data in/out ready iu sanity checks
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 8e453155d7f8dfa53863ba6f8da6c68f7c17ece4
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Fri Nov 15 10:04:11 2013 +0100

    uas: Add some data in/out ready iu sanity checks

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/uas.c b/drivers/usb/storage/uas.c
index f09205b..6208682 100644
--- a/drivers/usb/storage/uas.c
+++ b/drivers/usb/storage/uas.c
@@ -379,9 +379,19 @@ static void uas_stat_cmplt(struct urb *urb)
   uas_try_complete(cmnd, __func__);
   break;
  case IU_ID_READ_READY:
+  if (!cmdinfo->data_in_urb ||
+    (cmdinfo->state & DATA_IN_URB_INFLIGHT)) {
+   scmd_printk(KERN_ERR, cmnd, "unexpected read rdy\n");
+   break;
+  }
   uas_xfer_data(urb, cmnd, SUBMIT_DATA_IN_URB);
   break;
  case IU_ID_WRITE_READY:
+  if (!cmdinfo->data_out_urb ||
+    (cmdinfo->state & DATA_OUT_URB_INFLIGHT)) {
+   scmd_printk(KERN_ERR, cmnd, "unexpected write rdy\n");
+   break;
+  }
   uas_xfer_data(urb, cmnd, SUBMIT_DATA_OUT_URB);
   break;
  default:
-- 
1.7.1